docs/presidio_V2.md

```diff
@@ -0,0 +1,174 @@
+# Presidio Revamp (aka V2)
+
+As of March 2021, Presidio had undergo a revamp to a new version refereed to as **V2**.
+
+The main changes introduced in **V2** are:
+
+1. gRPC replaced with HTTP to allow more customizable APIs and easier debugging
+2. Presidio Anonymizer is Python based and pip installable.
+3. Focus on the Analyzer and Anonymizer services. Other services are deprecated and potentially be migrated over time to **V2** with the help of the community.
+4. Improved documentation, samples and build flows.
+5. Presidio Analyzer - removed use of templates and external recognizer store
+
+Notes:
+#TODO:check link update master branch update
+The legacy V1 code base will continue to be available under branch [V1](https://github.com/microsoft/presidio/tree/V1) but will no longer be officially supported.
+
+## API Changes
+
+The move from gRPC to HTTP based APIs included changes to the API requests.
+
+1. Change in payload - moving from structures to jsons.
+
+2. Removing templates from the API - includes flattening the json.
+3. Using snake_case instead of camel_case.
+
+Bellow is a detailed outline of all the changes done to the Analyzer and Anonymizer.
+
+### Analyzer API Changes
+
+#### Legacy json request (gRPC)
+
+```json
+{
+    "text": "My phone number is 212-555-5555",
+    "AnalyzeTemplateId": "1234",
+    "AnalyzeTemplate": {
+        "Fields": [
+            {
+                "Name": "PHONE_NUMBER",
+                "MinScore": "0.5"
+            }
+        ],
+        "AllFields": true,
+        "Description": "template description",
+        "CreateTime": "template creation time",
+        "ModifiedTime": "template modification time",
+        "Language": "fr",
+        "ResultsScoreThreshold": 0.5
+    }
+}
+```
+
+#### V2 json request (HTTP)
+
+```json
+{
+    "text": "My phone number is 212-555-5555",
+    "entities": ["PHONE_NUMBER"],
+    "language": "en",
+    "correlation_id": "213",
+    "score_threshold": 0.5,
+    "trace": true,
+    "remove_interpretability_response": true

```

Comment:
Whoever merges first should update this to return_decision_process 😊

Reply:
done